ksqldb-rest-app/src/main/java/io/confluent/ksql/api/auth/JaasPrincipal.java

```diff
@@ -15,18 +15,32 @@
 
 package io.confluent.ksql.api.auth;
 
+import java.nio.charset.StandardCharsets;
 import java.security.Principal;
+import java.util.Base64;
 import java.util.Objects;
 
 /**
  * Principal implementation created when authenticating with the JaasAuthProvider
  */
-class JaasPrincipal implements Principal {
+public class JaasPrincipal implements Principal {
 
   private final String name;
 
-  JaasPrincipal(final String name) {
+  private final String token;
+
+  public JaasPrincipal(final String name, final String password) {
     this.name = Objects.requireNonNull(name);
+    this.token = createToken(name, Objects.requireNonNull(password));
```

Comment:
Ok, I was hoping to pass the token directly to the constructor. But I see the password is already decoded from the JaasAuthProvider, so it make sense to pass the password. In this case, just keep the password in the field. I thought we were going to get access to the HTTP authorization header, which has the encoded string, and pass it to the Principal. 

The MDS API accept the user/pass too, so we can use that and avoid to encode the string there

Reply:
1b28358